src/main/scala/mesosphere/marathon/core/storage/simple/PersistenceStore.scala

```diff
@@ -0,0 +1,124 @@
+package mesosphere.marathon
+package core.storage.simple
+
+import akka.stream.scaladsl.{Flow, Source}
+import akka.util.ByteString
+import akka.{Done, NotUsed}
+import mesosphere.marathon.core.storage.simple.PersistenceStore.{Node, StoreOp}
+
+import scala.concurrent.Future
+import scala.util.Try
+
+/**
+  * An interface for a simple persistence store abstraction for the underlying Zookeeper store. Supported are basic
+  * create, read, update, delete and check methods along with transaction operations and Zookeeper specific children
+  * and sync methods.
+  */
+trait PersistenceStore {
+
+  /**
+    * A Flow for saving nodes to the store. It takes a stream of nodes and returns a stream of node keys
+    * that were successfully stored.
+    *
+    * By default persistent nodes with empty ACLs are created. If the path is nested parent nodes will be also
+    * created if they don't already exist. If there is already a node with the same path a [[org.apache.zookeeper.KeeperException.NodeExistsException]]
+    * is thrown.
+    *
+    * @return
+    */
+  def create: Flow[Node, String, NotUsed]
```

Comment:
I don't really get this one. Is it creating an empty node?

Reply:
No, it just returns a `Flow` from `Node -> String` which when materialized will create node(s) in ZK. You can visualize it like:
```
                                         Flow
                                   ------------
Node(path, data)     |   creating   |     path
------------------> |      node     | ---------->
                                  |      in ZK     |
                                  -------------
```